### PR TITLE
Drone support

### PR DESCRIPTION
**Pull request by maplain**
fixes: https://github.com/vmware/vic/issues/4971

### CODE REVIEW

**Comment by maplain** [comment]
still needs supervised-io to be turned on to generate .drone.sig and add secret for slack.

**Comment by jakedsouza** [comment]
Can you split vendor to a separate PR.

**Comment by maplain** [comment]
@jaked isn't it?

File: Makefile [hunk 1]
```diff
@@ -55,7 +55,7 @@ clean: ## Clean all modified files
 
 .PHONY: generate
 generate: ## run go generate
-	$(GO) generate ./cmd
+	$(GO) generate ./cmd/*
```
Comment (jakedsouza) [inline Makefile]: change to `$(GO) generate .` there are no generate tags in cmd

Comment (maplain) [inline Makefile]: run 'go generate ./cmd':

"can't load package: package github.com/supervised-io/kov/cmd: no buildable Go source files in /Users/fangyuanl/development/go/src/github.com/supervised-io/kov/cmd"

File: .drone.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  path: src/github.com/supervised-io/kov
+
+pipeline:
+  compile:
+    image: golang:1.8
+    pull: true
+    commands:
+      - "echo \"GOPATH: $GOPATH\""
+      - "echo \"pwd: $(pwd)\""
+      - go get -u github.com/golang/mock/mockgen
```
Comment (maplain) [inline .drone.yml]: I suppose the idea here is to install no more than what we needed in tasks to save time

File: .drone.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      - make test
+
+  build_dist:
+     image: golang:1.8.1-alpine
+     environment:
+       DEV: 0
+     pull: true
+     commands:
+       - apk add --no-cache git bash
+       - go get github.com/mitchellh/gox
```
Comment (maplain) [inline .drone.yml]: we don't seem to need gometalinter, glide, swagger here.

File: .drone.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+  slack:
+    image: plugins/slack
+    channel: vic-kov
+    username: drone
+    icon_emoji: ':drone:'
+    when:
+      repo: supervised-io/kov
+      event: [ push, tag ]
+      branch: [ master, releases/*, refs/tags/* ]
```
Comment (maplain) [inline .drone.yml]: 👌

File: .drone.yml [hunk 5]
```diff
@@ -0,0 +1,39 @@
+---
+workspace:
+  base: /go
+  path: src/github.com/supervised-io/kov
+
+pipeline:
+  compile:
+    image: golang:1.8
```
Comment (maplain) [inline .drone.yml]: 👌

File: Makefile (hunk 1 above)
Comment (casualjim) [inline Makefile]: it's like this ./cmd/...

File: .drone.yml (hunk 5 above)
Comment (casualjim) [inline .drone.yml]: use casualjim/builder

Comment (jakedsouza) [inline .drone.yml]: > use casualjim/builder

we would need casualjim/builder:1.8

**Comment by maplain** [comment]
casualjim/builder doesn't have make at my first try. I'm going to stick to golang image for 'make test' and casualjim/builder for build dist.

**Pull request merged**